app/Http/Controllers/ContactsController.php

```diff
@@ -570,4 +547,118 @@ public function archive(Request $request, Contact $contact)
             'is_active' => $contact->is_active,
         ];
     }
+
+    /**
+     * Display the list of contacts.
+     *
+     * @param Request $request
+     * @return array
+     */
+    public function list(Request $request)
+    {
+        $contactsCollection = collect([]);
+
+        $accountId = auth()->user()->account_id;
+
+        $user = $request->user();
+        $sort = $request->get('sort') ?? $user->contacts_sort_order;
+        $showDeceased = $request->get('show_dead');
+
+        if ($user->contacts_sort_order !== $sort) {
+            $user->updateContactViewPreference($sort);
+        }
+
+        $tags = null;
+        $url = '';
+        $count = 1;
+
+        $contacts = $user->account->contacts()->real();
+        $contacts = $contacts->active();
+
+        if ($request->get('no_tag')) {
+            // get tag less contacts
+            $contacts = $contacts->tags('NONE');
+        } elseif ($request->get('tag1')) {
+            // get contacts with selected tags
+            $tags = collect();
+
+            while ($request->get('tag'.$count)) {
+                $tag = Tag::where('account_id', $accountId)
+                    ->where('name_slug', $request->get('tag'.$count))
+                    ->get();
+
+                if (! ($tags->contains($tag[0]))) {
+                    $tags = $tags->concat($tag);
+                }
+
+                $url = $url.'tag'.$count.'='.$tag[0]->name_slug.'&';
+
+                $count++;
+            }
+            if (! is_null($tags)) {
+                $contacts = $contacts->tags($tags);
+            }
+        }
+
+        // filter out deceased if necessary
+        if ($showDeceased != 'true') {
+            $contacts = $contacts->where('is_dead', 0);
```

Comment:
You may use the `scopeAlive` you've added on `Contact` model:
```suggestion
            $contacts = $contacts->alive();
```